src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -0,0 +1,692 @@
+			String type = InventoryType.main.toString();
+			try {
+				Page<MedicalInventory> medInventorypage = medicalInventoryManager.getMedicalInventoryByParamsPageable(dateFrom, dateTo, state, type, page,
+					pageSize);
+				inventoryList = medInventorypage.getContent();
+			} catch (OHServiceException e) {
+				OHServiceExceptionUtil.showMessages(e);
+			}
+		}
+
+		public Class< ? > getColumnClass(int c) {
+			if (c == 0) {
+				return String.class;
+			} else if (c == 1) {
+				return String.class;
+			} else if (c == 2) {
+				return String.class;
+			} else if (c == 3) {
+				return String.class;
+			}
+			return null;
+		}
+
+		public int getRowCount() {
+			if (inventoryList == null) {
+				return 0;
+			}
+			return inventoryList.size();
+		}
+
+		public String getColumnName(int c) {
+			return columsNames[c];
+		}
+
+		public int getColumnCount() {
+			return columsNames.length;
+		}
+
+		public Object getValueAt(int r, int c) {
+			MedicalInventory medInvt = inventoryList.get(r);
+			if (c == -1) {
+				return medInvt;
+			} else if (c == 0) {
+				return medInvt.getInventoryReference();
+			} else if (c == 1) {
+				return medInvt.getInventoryDate().format(DATE_TIME_FORMATTER);
+			} else if (c == 2) {
+				return medInvt.getStatus();
+			} else if (c == 3) {
+				return medInvt.getUser();
+			}
+			return null;
+		}
+
+		@Override
+		public boolean isCellEditable(int arg0, int arg1) {
+			return false;
+		}
+
+	}
+
+	public String formatDateTime(GregorianCalendar time) {
+		SimpleDateFormat format = new SimpleDateFormat("dd/MM/yy"); //$NON-NLS-1$
+		return format.format(time.getTime());
+	}
+
+	private JComboBox<String> getComboBox() {
+		if (statusComboBox == null) {
+			statusComboBox = new JComboBox<String>();
+			statusComboBox.addItem(MessageBundle.getMessage("angal.common.all.txt"));
+			List<String> statusList = medicalInventoryManager.getStatusList();
+			for (String status : statusList) {
+				statusComboBox.addItem(status);
+			}
+			statusComboBox.addActionListener(actionEvent -> {
+				totalRows = medicalInventoryManager.getInventoryCount(InventoryType.main.toString());
+				startIndex = 0;
+				int page = 1;
+				previous.setEnabled(false);
+				if (totalRows <= PAGE_SIZE) {
+					next.setEnabled(false);
+				} else {
+					next.setEnabled(true);
+				}
+				jTableInventory.setModel(new InventoryBrowsingModel(page, PAGE_SIZE));
+				initialisePagesCombo(totalRows);
+			});
+		}
+		return statusComboBox;
+	}
+
+	private JLabel getStatusLabel() {
+		if (statusLabel == null) {
+			statusLabel = new JLabel(MessageBundle.getMessage("angal.inventory.status.label"));
+			statusLabel.setHorizontalAlignment(SwingConstants.RIGHT);
+		}
+		return statusLabel;
+	}
+
+	public void initialisePagesCombo(int total_rows) {
+		int j = 0;
+
+		// if totalRows = 0 we have at least 1 page
+		int totalPages = Math.max(1, (int) Math.ceil((double) totalRows / PAGE_SIZE));
```

Comment:
can't see how this works as the method parameter is `total_rows` and here it is `totalRows`